### ISSUE TITLE
[Enhancement]: Support for cluster-level Performance Insights in Aurora clusters

### ISSUE THREAD

**Issue by yuto-ohta** [hashicorp/terraform-provider-aws#40697]
##### Context
As of September 2024, AWS announced support for cluster-level Performance Insights configuration for Aurora clusters (announced [here](https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-rds-performance-insights-supports-aurora-cluster-level-configuration/?nc1=h_ls)). This allows users to enable Performance Insights at the cluster level rather than configuring it for individual instances.

##### Current Behavior
Currently, in the `aws_rds_cluster` resource, the `performance_insights_enabled` attribute is only supported for RDS Clusters, not for Aurora clusters. This means we cannot leverage the new cluster-level Performance Insights feature for Aurora through Terraform.

##### Requested Change
Please add support for the `performance_insights_enabled` attribute in `aws_rds_cluster` resource when using Aurora. This would allow users to enable and manage Performance Insights at the cluster level for Aurora clusters, aligning with AWS's latest feature support.

### Affected Resource(s) and/or Data Source(s)

aws_rds_cluster

### Potential Terraform Configuration

```terraform
resource "aws_rds_cluster" "example" {
  cluster_identifier           = "example"
  engine                       = "aurora-mysql"
  performance_insights_enabled = true
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-rds-performance-insights-supports-aurora-cluster-level-configuration/?nc1=h_ls

### Would you like to implement a fix?

No

**Comment by hjoshi123** [comment]
@yuto-ohta can I work on this?

**Comment by hjoshi123** [comment]
@yuto-ohta Looking at the code there's already support for this for creating, updating etc.
```
"performance_insights_enabled": {
				Type:     schema.TypeBool,
				Optional: true,
			},
```

**Comment by yuto-ohta** [comment]
@hjoshi123 

> can I work on this?

Thank you for your interest! 

> Looking at the code there's already support for this for creating, updating etc.

Thank you for checking the code. While the attribute `performance_insights_enabled` does exist in the schema, it was implemented specifically for RDS Clusters in PR #29415, not for Aurora clusters. This is also reflected in our documentation which states:

`performance_insights_enabled - (Optional) Enables Performance Insights for the RDS Cluster`
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#performance_insights_enabled-1

The current issue is about adding support for this attribute specifically for Aurora clusters.

**Comment by yuto-ohta** [comment]
@hjoshi123 
Thank you for the detailed investigation and for pointing to the AWS API documentation. You are absolutely right - I checked and confirmed that `performance_insights_enabled` does work for Aurora clusters as well.

This was my misunderstanding based on the current documentation. Thanks for catching this!
